metadata/src/utils/validation.rs

```diff
@@ -195,11 +212,38 @@ pub fn get_type_hash(
 fn get_extrinsic_hash(
     registry: &PortableRegistry,
     extrinsic: &ExtrinsicMetadata,
+    specific_pallets: &Option<Vec<String>>,
```

Comment:
I reckon that it would be cleaner to change this to (if possible):
```suggestion
    specific_pallets: Option<&[String]>,
```